File: app/src/main/java/com/yuyin/demo/SpeechTextAdapter.java
package com.yuyin.demo;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


import java.util.List;

public class SpeechTextAdapter extends RecyclerView.Adapter<SpeechTextAdapter.ViewHolder> {
    private List<SpeechText> dataList;

    public SpeechTextAdapter(List<SpeechText> data) {
        dataList = data;
    }

    @NonNull
    @Override
    public SpeechTextAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.text_item, parent, false);
        return new SpeechTextAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SpeechTextAdapter.ViewHolder holder, int position) {
        SpeechText speechText = dataList.get(position);
        holder.getSpeech_view().setText(speechText.getText());
    }


    @Override
    public int getItemCount() {
        return dataList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView speech_view;
        public ViewHolder(View view) {
            super(view);
            this.speech_view = (TextView) view.findViewById(R.id.speechText);
        }
        public TextView getSpeech_view() {
            return speech_view;
        }
    }



}

File: app/src/main/java/com/yuyin/demo/SpeechTextAdapter.kt
package com.yuyin.demo

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class SpeechTextAdapter(private val dataList: List<SpeechText>) :
    RecyclerView.Adapter<SpeechTextAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.text_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val speechText = dataList[position]
        holder.speechView.text = speechText.text
    }

    override fun getItemCount(): Int {
        return dataList.size
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val speechView = view.findViewById<View>(R.id.speechText) as TextView

    }
}